File: 20-valid-parentheses/20. Valid Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> mystack;

        for(char c : s)
        {
            if(c=='('|| c== '{' || c=='[')
            {
                mystack.push(c);
            }
            else 
            {
                if(mystack.empty())
                {
                    return false;
                }
                char top=mystack.top();
                mystack.pop();
                if(c==')'&& top!='('
                   || c=='}'&& top!='{'
                   || c==']'&& top!='[')
                   {
                    return false;
                   }
            }
        }
        return  mystack.empty();
    }
};

File: 20-valid-parentheses/20. Valid Parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack <Character> mystack =new Stack<>();
        for(char c:s.toCharArray()){
            if(c=='('||c=='{'||c=='[')
            {
                mystack.push(c);
            }
            else{
                if(mystack.isEmpty())
                {
                    return false;
                }
                char top=mystack.pop();
                if(c==')'&& top!='('
                    || c=='}'&& top!='{'
                    || c==']'&& top!='[')
                    {
                        return false;
                    }
            }
        }
        return mystack.isEmpty();
    }
}